packages/g/gtest/xmake.lua

```diff
@@ -23,16 +35,39 @@ package("gtest")
                 add_includedirs("googletest/include", "googletest")
                 add_headerfiles("googletest/include/(**.h)")
 
+            target("gtest_main")
+                set_kind("static")
+                set_languages("cxx11")
+                set_default(]] .. tostring(package:config("gtest_main")) .. [[)
+                add_files("googletest/src/gtest_main.cc")
+                add_includedirs("googletest/include", "googletest")
+                add_headerfiles("googletest/include/(**.h)")
+
             target("gmock")
                 set_kind("static")
                 set_languages("cxx11")
+                set_default(]] .. tostring(package:config("gmock")) .. [[)
                 add_files("googlemock/src/gmock-all.cc")
                 add_includedirs("googlemock/include", "googlemock", "googletest/include", "googletest")
                 add_headerfiles("googlemock/include/(**.h)")
         ]])
         import("package.tools.xmake").install(package)
     end)
 
+    on_load(function (package)
+        package:set("links", {})
+
+        if package:config("main") then
+            package:add("links", "gtest_main")
+        end
+
+        if package:config("gmock") then
+            package:add("links", "gmock")
+        end
+
+        package:add("links", "gtest")
+    end)
```

Comment:
we need not this on_load, because you have set them in on_install/set_default. 

please remove it.